dist/sync-promise-amd.js

```diff
@@ -26,58 +29,66 @@ function SyncPromise(fn) {
     self.c = null; // Release memory.
   }
   function resolve(val) {
-    if (!self.c) {
-      // Already resolved, do nothing.
+    if (syncResolved) {
+      syncRes = true;
+      val = SyncPromise.resolve(val);
+    }
+    if (!self.c || syncErr) {
+      // Already resolved (or will be resolved), do nothing.
     } else if (isPromise(val)) {
-      val.then(resolve).catch(reject);
+      addReject(val.then(resolve), reject);
     } else {
       transist(val, FULFILLED);
     }
   }
   function reject(reason) {
-    if (!self.c) {
-      // Already resolved, do nothing.
+    if (syncResolved) {
+      syncErr = true;
+      reason = SyncPromise.reject(reason);
+    }
+
+    if (!self.c || syncRes) {
+      // Already resolved (or will be resolved), do nothing.
     } else if (isPromise(reason)) {
-      reason.then(reject).catch(reject);
+      addReject(reason.then(reject), reject);
     } else {
       transist(reason, REJECTED);
     }
   }
-  fn(resolve, reject);
+  try {
+    fn(resolve, reject);
+  } catch (err) {
+    reject(err);
+  }
   syncResolved = false;
 }
 
 var prot = SyncPromise.prototype;
 
-prot.then = function(cb) {
+prot.then = function(cb, errBack) {
   var self = this;
-  if (self.a) { // Promise has been resolved synchronously
-    throw new Error('Can not call then on synchonously resolved promise');
-  }
   return new SyncPromise(function(resolve, reject) {
+    var rej = typeof errBack === 'function' ? errBack : reject;
     function settle() {
       try {
-        resolve(cb(self.v));
+        resolve(cb ? cb(self.v) : self.v);
       } catch(e) {
-        reject(e);
+        rej(e);
       }
     }
     if (self.s === FULFILLED) {
       settle();
     } else if (self.s === REJECTED) {
-      reject(self.v);
+      rej(self.v);
     } else {
       self.c[FULFILLED].push(settle);
-      self.c[REJECTED].push(reject);
+      self.c[REJECTED].push(rej);
     }
   });
 };
 
 prot.catch = function(cb) {
   var self = this;
-  if (self.a) { // Promise has been resolved synchronously
-    throw new Error('Can not call catch on synchonously resolved promise');
-  }
```

Comment:
Why did you remove the error checking here? It's quite useful to ensure that a promise is not accidentally resolved synchronously.